wallet-next/src/bin/pwalletd.rs

```diff
@@ -1,22 +1,210 @@
-use anyhow::Result;
+#[allow(clippy::clone_on_copy)]
+use anyhow::{Context, Result};
+use penumbra_crypto::keys::FullViewingKey;
+use penumbra_proto::client::oblivious::{
+    oblivious_query_client::ObliviousQueryClient, AssetListRequest, ChainParamsRequest,
+    CompactBlockRangeRequest,
+};
 use sqlx::sqlite::SqlitePool;
 use std::env;
+use tonic::transport::{Channel, Server};
+use tracing::instrument;
 
 use penumbra_wallet_next::Storage;
 
+use std::path::PathBuf;
+
+use directories::ProjectDirs;
+
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+#[structopt(
+    name = "pwalletd",
+    about = "The Penumbra wallet daemon.",
+    version = env!("VERGEN_GIT_SEMVER"),
+)]
+struct Opt {
+    /// Command to run.
+    #[structopt(subcommand)]
+    cmd: Command,
+}
+
+#[derive(Debug, StructOpt)]
+enum Command {
+    /// Start running the wallet daemon.
+    Init {
+        /// The path used to store the SQLite state database.
+        #[structopt(short, long)]
+        sqlite_path: PathBuf,
+        /// The full viewing key
+        #[structopt(short, long)]
+        fvk: String,
+    },
+    Start {
+        /// Bind the services to this host.
+        #[structopt(short, long, default_value = "127.0.0.1")]
+        host: String,
+        /// Bind the wallet gRPC server to this port.
+        #[structopt(short, long, default_value = "26668")]
+        grpc_port: u16,
+        /// The address of the pd+tendermint node.
+        #[structopt(short, long, default_value = "testnet.penumbra.zone")]
+        node: String,
+        /// The port to use to speak to pd's light wallet server.
+        #[structopt(short, long, default_value = "26666")]
+        oblivious_query_port: u16,
+        /// The port to use to speak to pd's thin wallet server.
+        #[structopt(short, long, default_value = "26667")]
+        specific_query_port: u16,
```

Comment:
These options should be folded together into a single `pd_port` with default value `8080`, now that #730 unified them into a single endpoint.